api/internal/localizer/localizer_test.go

```diff
@@ -477,39 +606,25 @@ paths:
 func TestLocalizeValidators(t *testing.T) {
 	fSys := makeMemoryFs(t)
 	kustAndPlugin := map[string]string{
-		"kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1beta1
+		"kustomization.yaml": fmt.Sprintf(`apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 validators:
-- |-
+- |
   apiVersion: builtin
   kind: ReplacementTransformer
   metadata:
     name: replacement
   replacements:
-  - source:
-      kind: ConfigMap
-      fieldPath: metadata.name
-    targets:
-    - select:
-        kind: ConfigMap
-      fieldPaths:
-      - metadata.name
+  - %s
 - replacement.yaml
-`,
-		"replacement.yaml": `apiVersion: builtin
+`, strings.ReplaceAll(replacement, "\n", "\n    ")),
+		"replacement.yaml": fmt.Sprintf(`apiVersion: builtin
 kind: ReplacementTransformer
 metadata:
   name: replacement-2
 replacements:
-- source:
-    kind: Secret
-    fieldPath: data.USER_NAME
-  targets:
-  - select:
-      kind: Secret
-    fieldPaths:
-    - data.USER_NAME
-`,
+- %s
+`, strings.ReplaceAll(replacement, "\n", "\n  ")),
```

Comment:
why are you doing `strings.ReplaceAll`? Can we just add whatever newlines or whitespace you want in the string literals themselves?

Reply:
I replaced the `strings.ReplaceAll` occurrences with string literals.

I used `strings.ReplaceAll` because I was concerned the inline patch and replacements were hindering the readability of the tests. The different occurrences require different spacing depending on the configuration before the occurrence. It didn't make sense to create a string literal for each spacing requirement, so I modified the spacing on a single literal to fit each use case.